tests/unit/server/api/test_subscriptions.py

```diff
@@ -0,0 +1,268 @@
+            query=self.QUERY,
+            variables=variables,
+            operation_name="ChatCompletionSubscription",
+        ) as subscription:
+            with use_cassette(
+                Path(__file__).parent / "cassettes/test_subscriptions/"
+                "TestChatCompletionSubscription.test_openai_text_response_emits_expected_payloads_and_records_expected_span[sqlite].yaml",
+                decode_compressed_response=True,
+                before_record_request=remove_all_vcr_request_headers,
+                before_record_response=remove_all_vcr_response_headers,
+            ):
+                payloads = [payload["chatCompletion"] async for payload in subscription.stream()]
+
+        # check subscription payloads
+        assert payloads
+        assert (last_payload := payloads.pop())["__typename"] == "FinishedChatCompletion"
+        assert all(payload["__typename"] == "TextChunk" for payload in payloads)
+        response_text = "".join(payload["content"] for payload in payloads)
+        assert "france" in response_text.lower()
+        subscription_span = last_payload["span"]
+        span_id = subscription_span["id"]
+
+        # query for the span via the node interface to ensure that the span
+        # recorded in the db contains identical information as the span emitted
+        # by the subscription
+        data = await gql_client.execute(
+            query=self.QUERY, variables={"spanId": span_id}, operation_name="SpanQuery"
+        )
+        span = data["span"]
+        assert span == subscription_span
+
+        # check attributes
+        assert span.pop("id") == span_id
+        assert span.pop("name") == "Chat Completion"
+        assert span.pop("statusCode") == "OK"
+        assert not span.pop("statusMessage")
+        assert span.pop("startTime")
+        assert span.pop("endTime")
+        assert isinstance(span.pop("latencyMs"), float)
+        assert span.pop("parentId") is None
+        assert span.pop("spanKind") == "llm"
+        assert (context := span.pop("context")).pop("spanId")
+        assert (attributes := dict(flatten(json.loads(span.pop("attributes")))))
+        assert context.pop("traceId")
+        assert not context
+        assert span.pop("metadata") is None
+        assert span.pop("numDocuments") is None
+        assert isinstance(token_count_total := span.pop("tokenCountTotal"), int)
+        assert isinstance(token_count_prompt := span.pop("tokenCountPrompt"), int)
+        assert isinstance(token_count_completion := span.pop("tokenCountCompletion"), int)
+        assert token_count_prompt > 0
+        assert token_count_completion > 0
+        assert token_count_total == token_count_prompt + token_count_completion
+        assert (input := span.pop("input")).pop("mimeType") == "json"
+        assert (input_value := input.pop("value"))
+        assert not input
+        assert "api_key" not in input_value
+        assert "apiKey" not in input_value
```

Comment:
is there impl logic is to guard against this?

Reply:
No, just threw it in there for good measure